### ISSUE TITLE
How to Use HY3D BPT in ComfyUI"

### ISSUE THREAD

**Issue by lcwzx** [kijai/ComfyUI-Hunyuan3DWrapper#80]
How to Use HY3D BPT in ComfyUI"

**Comment by Easymode-ai** [comment]
add the `Hy3D BPT` node after the `Hy3D PostProcess Mesh` or `Remesh` set to around 10-50k faces, and then continue on to export mesh/preview mesh.

**Comment by asifagha** [comment]
@Easymode-ai i have tried that but I noticed the generated mesh is always worse and take 10X the time to generate. 
It seems that it can't exceed the 10K samples. 

got prompt
/home/utest/miniconda3/envs/3dwrapper_v3env/lib/python3.10/site-packages/x_transformers/x_transformers.py:435: FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
  @autocast(enabled = False)
/home/utest/miniconda3/envs/3dwrapper_v3env/lib/python3.10/site-packages/x_transformers/x_transformers.py:461: FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
  @autocast(enabled = False)
[2025-02-25 18:59:21,403] [INFO] [real_accelerator.py:222:get_accelerator] Setting ds_accelerator to cuda (auto detect)
Point cloud encoder: miche-256-feature | freeze: False
{'model': {'target': 'hy3dgen.shapegen.bpt.miche.michelangelo.models.tsal.asl_pl_module.AlignedShapeAsLatentPLModule', 'params': {'shape_module_cfg': {'target': 'hy3dgen.shapegen.bpt.miche.michelangelo.models.tsal.sal_perceiver.AlignedShapeLatentPerceiver', 'params': {'num_latents': 256, 'embed_dim': 64, 'point_feats': 3, 'num_freqs': 8, 'include_pi': False, 'heads': 12, 'width': 768, 'num_encoder_layers': 8, 'num_decoder_layers': 16, 'use_ln_post': True, 'init_scale': 0.25, 'qkv_bias': False, 'use_checkpoint': True}}, 'aligned_module_cfg': {'target': 'hy3dgen.shapegen.bpt.miche.michelangelo.models.tsal.clip_asl_module.CLIPAlignedShapeAsLatentModule', 'params': {'clip_model_version': './checkpoints/clip/clip-vit-large-patch14'}}, 'loss_cfg': {'target': 'hy3dgen.shapegen.bpt.miche.michelangelo.models.tsal.loss.ContrastKLNearFar', 'params': {'contrast_weight': 0.1, 'near_weight': 0.1, 'kl_weight': 0.001}}, 'optimizer_cfg': {'optimizer': {'target': 'torch.optim.AdamW', 'params': {'betas': [0.9, 0.99], 'eps': 1e-06, 'weight_decay': 0.01}}, 'scheduler': {'target': 'hy3dgen.shapegen.bpt.miche.michelangelo.utils.trainings.lr_scheduler.LambdaWarmUpCosineFactorScheduler', 'params': {'warm_up_steps': 5000, 'f_start': 1e-06, 'f_min': 0.001, 'f_max': 1.0}}}}}}
 custom path :
/home/utest/comfy/3dwrapper_v3/custom_nodes/ComfyUI-Hunyuan3DWrapper


 custom path :
/home/utest/comfy/3dwrapper_v3/custom_nodes/ComfyUI-Hunyuan3DWrapper


 custom path :
/home/utest/comfy/3dwrapper_v3/custom_nodes/ComfyUI-Hunyuan3DWrapper


 custom path :
/home/utest/comfy/3dwrapper_v3/custom_nodes/ComfyUI-Hunyuan3DWrapper


/home/utest/comfy/3dwrapper_v3/custom_nodes/ComfyUI-Hunyuan3DWrapper/hy3dgen/shapegen/bpt/bpt-8-16-500m.pt
Process: 0:   0%|                                                                                                 | 0/10000 [00:00<?, ?it/s]/home/utest/miniconda3/envs/3dwrapper_v3env/lib/python3.10/contextlib.py:103: FutureWarning: `torch.backends.cuda.sdp_kernel()` is deprecated. In the future, this context manager will be removed. Please see `torch.nn.attention.sdpa_kernel()` for the new context manager, with updated signature.
  self.gen = func(*args, **kwds)
Hy3DMeshInfo: Mesh has 758 vertices and 1520 faces 


<img width="1801" alt="Image" src="https://github.com/user-attachments/assets/a2743ba3-72ed-4557-8f42-67da3bcbe72d" />

**Comment by Easymode-ai** [comment]
Ah yes this is a weird issue with the faces generated I think .

I released an independent comfyui-bpt on my git which includes its own face reduction code as part of the node itself and I’m getting better results with that now, no more big sections cut out etc . Try it and let me know if that works better for you .
